applications/ThresholdSignature.md

```diff
@@ -0,0 +1,300 @@
+# Threshold Signature Implementation
+
+- **Team Name:** Rui Morais
+- **Payment Address:** 14bBGQFAgKqdbGVDSWkm6dA8ZQzt9GxGSTALrD8SeafWW9gL (USDC)
+- **[Level](https://github.com/w3f/Grants-Program/tree/master#level_slider-levels):** 1
+
+## Project Description :page_facing_up:
+
+### General Overview
+
+This document outlines the proposal for implementing [Olaf](https://eprint.iacr.org/2023/899), the first FROST based protocol proven secure without relying on the [Algebraic Group Model](https://eprint.iacr.org/2017/620.pdf) (AGM) or an idealized key generation protocol. It is composed of:
+- [FROST3](https://eprint.iacr.org/2022/550), the most efficient known variant of [FROST](https://eprint.iacr.org/2020/852), which is a threshold signature protocol based on the [Schnorr signature scheme](https://link.springer.com/article/10.1007/BF00196725) (for simplicity, we refer to FROST3 as FROST in this document, since the core functionality is the same).
+- [SimplPedPoP](https://eprint.iacr.org/2023/899), a Distributed Key Generation (DKG) protocol used to generate the group key to sign in FROST. It is a simplification of [PedPoP](https://eprint.iacr.org/2020/852), the DKG used in the original FROST paper, which is the [Pedersen's DKG](https://link.springer.com/content/pdf/10.1007/3-540-46766-1_9.pdf) with Proofs of Possession (PoP) of the generated secret. All of them have as the fundamental building block the [Shamir's secret sharing scheme](https://web.mit.edu/6.857/OldStuff/Fall03/ref/Shamir-HowToShareASecret.pdf).
```

Comment:
SimplePedPop sounds fine.  I never looked at PedPoP really, only the original Pedersen DKG.  Again any comment @elizabeth-crites?

We'd simply output a hash for the equality check algorithm, so then the user must handle the final abort if that fails?

Reply:
Ahh SimplePedPoP does not encapsulate encrypting the key shares.  Does PedPoP do so?

I'll think about if there is a way to abstract this, but maybe SimplePedPoP works well for single user use cases.

Reply:
PedPoP does not encapsulate encrypting key shares either. That is why they are implementing EncPedPop [here](https://github.com/BlockstreamResearch/bip-frost-dkg), which is a wrapper around SimplPedPop. And they recommend it even for applications that support secure channels, due to its low overhead. Should we do the same?

Reply:
It adds an extra round because you must know to whom you encrypt.  As EncPedPop is a wrapper around SimplePedPoP then I'd say do SimplePedPoP first, and we can think about the wrapper later.  It'll maybe require cleaning up the aead module, so no reason to block anything:  https://github.com/w3f/schnorrkel/blob/master/src/aead.rs